website/source/docs/secrets/jwt/index.html.md

```diff
@@ -0,0 +1,277 @@
+
+This page will show a quick start for this backend. For detailed documentation on every path, use `vault path-help` after mounting the backend.
+
+## Algorithms
+
+### RSA 
+* RS256
+* RS384
+* RS512
+
+These require a RSA private/public keypair for signing and verification.
+
+### ECDSA
+* EC256
+* EC384
+* EC512
+
+These require an ECDSA private/public keypair for signing and verification.
+
+### HMAC
+* HS256
+* HS384
+* HS512
+
+These require a shared secret for signing and verification.
+
+
+## Quick Start
+
+The first step to using the jwt backend is to mount it.
+Unlike the `generic` backend, the `jwt` backend is not mounted by default.
+
+```text
+$ vault mount jwt
+Successfully mounted 'jwt' at 'jwt'!
+```
+
+The next step is to configure a role. A role is a logical name that maps
+to a few settings used to generated the tokens. For example, lets create
+a "webauth" role:
+
+```text
+$ vault write jwt/roles/webauth \
+    name=webauth \
+    algorithm=RS256
+```
+
+This path will create a named role along with the algorithm used
+to sign the token.
+
+The next step is to configure the key for the role. Each role requires a secret or
+a private key to be associated against it. For example, lets configure a private
+key against the "webauth" role.
+
+```text
+$ vault write jwt/config/webauth \
+    key=@/path/to/private.key
+```
+
+Generating a token requires passing of additional information so we use the
+"jwt/issue/ROLE" path.
+
+```text
+$ vault write jwt/issue/webauth \
+    iss="Vault" \
+    aud="Vault Client" \
+    exp="1538096292" \
+    claims=@extra.json
+```
+
+The issue path has several 
+
+## API
+
+### /jwt/roles/
+#### POST
+
+<dl class="api">
+  <dt>Description</dt>
+  <dd>
+    Creates or updates a named role.
+  </dd>
+
+  <dt>Method</dt>
+  <dd>POST</dd>
+
+  <dt>URL</dt>
+  <dd>`/jwt/roles/<name>`</dd>
+
+  <dt>Parameters</dt>
+  <dd>
+    <ul>
+      <li>
+        <span class="param">algorithm</span>
+        <span class="param-flags">required</span>
+        The algorithm used by JWT to sign the token.
+      </li>
+    </ul>
+  </dd>
+
+  <dt>Returns</dt>
+  <dd>
+    A `204` response code.
+  </dd>
+</dl>
+
+#### GET
+
+<dl class="api">
+  <dt>Description</dt>
+  <dd>
+    Queries a named role.
+  </dd>
+
+  <dt>Method</dt>
+  <dd>GET</dd>
+
+  <dt>URL</dt>
+  <dd>`/jwt/roles/<name>`</dd>
+
+  <dt>Parameters</dt>
+  <dd>
+    None
+  </dd>
+
+  <dt>Returns</dt>
+  <dd>
+
+    ```javascript
+    {
+        "data": {
+            "algorithm": "..."
+        }
+    }
+    ```
+
+  </dd>
+</dl>
+
+#### DELETE
+
+<dl class="api">
+  <dt>Description</dt>
+  <dd>
+    Deletes a named role.
+  </dd>
+
+  <dt>Method</dt>
+  <dd>DELETE</dd>
+
+  <dt>URL</dt>
+  <dd>`/jwt/roles/<name>`</dd>
+
+  <dt>Parameters</dt>
+  <dd>
+    None
+  </dd>
+
+  <dt>Returns</dt>
+  <dd>
+    A `204` response code.
+  </dd>
+</dl>
+
+
+### /jwt/config/
+#### POST
+
+<dl class="api">
+  <dt>Description</dt>
```

Comment:
The docs here don't seem to match `jwt/config` from the examples. Copy/paste error?